app/components/questions/BelongingRates.js

```diff
@@ -49,18 +49,28 @@ function transformData(response, error, selectedArea) {
       valueFormatted: Formatter.percentWithOneDecimal(1 - areaData.belonging_rate_2018),
       color: Color.Salmon
     }
-    ]
+  ]
 
+  const dataFlag = (() => {
+  if (areaData.belongingRateFormatted==="NaN%")
+    return 1
+  else
+    return 0
+  })();
   
   return {
     chartData: data,
     areaData,
+    dataFlag,
```

Comment:
See comment below, for something like this, we can check it when the component renders, rather than adding it to the data transform.

Also, the syntax here of making an anonymous function and calling it is unnecessary. If we did want to add a new field, especially a boolean, we could do it like this:

```js
const missingData = !areaData.belonging_rate_2018 && areaData.belonging_rate_2018 !== 0;
...
return {
  chartData: data,
  areaData,
  missingData,
}
```

Reply:
If we check for the `missingData` in the component when it renders, would I have to pass `areaData` to the piechart as well? That was the only issue I had, considering the piechart got `data` but our area selector is passed `areaData`, which have their own structures

Reply:
I think I've solved my issue, I wasn't aware that `areaData.rate_of_belonging_2018` and `data.value` were essentially equivalent

Reply:
Ah yes, I forgot about the pie chart. It looks like what you wrote should work, but the website build fails. I'll open another thread for that.